tests/v2/validation/rbac/rbac_additional_test.go

```diff
@@ -57,149 +57,149 @@ func (rb *RBACAdditionalTestSuite) SetupSuite() {
 	require.NoError(rb.T(), err)
 }
 
-func (rb *RBACAdditionalTestSuite) ValidateAddStdUserAsProjectOwner() {
+func (rb *RBACAdditionalTestSuite) validateAddStdUserAsProjectOwner(t *testing.T) {
```

Comment:
I don't think you need `(rb *RBACAdditionalTestSuite)` anymore here since we pass testing.T now

Reply:
We can't remove that as we are using the objects from **RBACAdditionalTestSuite** struct.

Reply:
So if this will stay a member function, then there is no point in passing the testing.T it's very redundant. Thinking about it more, these are subtests in the same ilk of https://github.com/rancher/rancher/blob/64aba5d6c6d516640143d96679f3bd99dfd432d1/tests/v2/validation/upgrade/workload_test.go#L60

1. Maybe we should have these funcs have test(lowercase) prefixes. We can update our guidelines to make this clear for test type of sub test cases.
2. I think a lot of the member attributes/fields should be parameters, rather than update existing member fields. It makes this super hard to read, and I think one of the main issues.

Let me know what everyone thinks.

Reply:
I agree, this version makes the `testing.T` redundant, and the suite struct already has it. 

It makes sense to me to have subtests, in my opinion, subtests should mainly rely on what tests have, not on what the suite structure has. I think these can be private methods/subtests of the suite struct if they use parameters to customize the subtest just like @igomez06 's (2) point.

Reply:
I agree, that it is redundant and un required change, when we are wanting to passing testing.T when the test itself is implemented as a member function.
```
Maybe we should have these funcs have test(lowercase) prefixes. We can update our guidelines to make this clear for test type of sub test cases.
```
I'm ok with changing this to be a member function by adding `test` prefix as it avoids redundancy.

```
I think a lot of the member attributes/fields should be parameters, rather than update existing member fields. It makes this super hard to read, and I think one of the main issues.
```

The problem with using the parameter based approach is that we intend to utilize the struct fields throughout the entire testing process. Modifying the struct's fields directly is preferable because it allows us to avoid recreating standard clients and users repeatedly, which would introduce unnecessary redundancy, particularly since our tests depend on the outcomes of previously executed ones.

Reply:
`The problem with using the parameter based approach is that we intend to utilize the struct fields throughout the entire testing process. Modifying the struct's fields directly is preferable because it allows us to avoid recreating standard clients and users repeatedly, which would introduce unnecessary redundancy, particularly since our tests depend on the outcomes of previously executed ones.`

I don't know if I agree with this, and maybe I just need to understand better. Take for example rb.standardUserCOProject in the additional tests. Every time this parameter is used it is create a whole new project, except for this function `validateAddCMAsProjectOwner` so why doesn't this function just take a project parameter, so we know what project we are testing against. 

Or rb.additionalUser

We create this at the top test level so why is it a member function? The functions should take it as a parameter, so we know exactly the additional users we create. The things we create in the setup makes sense, since it's being use throughout the whole life cycle of the test. But these don't ... Let me know what you think.

Reply:
Sorry should've clarified, was taking rbac_test.go into consideration. You're right about the rb.standardUserCOProject, in additional tests that its being created for each of these tests independently. but the rb.additionalUser is the same user, that is used in all the tests. It is a bit different for each of these rbac tests and how each tests works independently or dependent on the previous tests. May be we can discuss about this and can come to a conclusion on what might be efficient here.

Reply:
Sure totally agree Let's set up a meeting and talk about it, that way we can all have an agreement, and move on